src/UseCases/DataAccessInterfaces/VendorRepository.java

```diff
@@ -19,5 +19,40 @@
      */
     IVendor getVendor(String id);
 
+    /**
+     *  A method that returns the authentication token used by the vendor
+     *
+     * @param id The associated id of the vendor.
+     * @return Return the authentication token used by the vendor.
+     */
+    String getAuthenticationToken(String id);
+
+    /**
+     *  A method that returns the id of a vendor from an authentication token.
+     *
+     * @param token The authentication token used by a vendor.
+     * @return Return the associated id of the vendor.
+     */
+    String getUserIdFromToken(String token);
+
+    /**
+     *  A method that returns the vendor from an authentication token.
+     *
+     * @param token The authentication token used by a vendor.
+     * @return Return the associated vendor.
+     */
+    IVendor getVendorFromToken(String token);
+
+    /**
+     *  A method that returns whether a vendor's authentication token is currently valid.
+     *
+     * @param token The authentication token used by a vendor.
+     * @return Return whether a vendor's authentication token is currently valid.
+     */
+    Boolean isTokenValid(String token);
+    /**
+    * Method for saving a vendor into the repository
+    * @param vendor the vendor object being saved to the repository
+    */
```

Comment:
This is an outdated version of vendor repository, please check the methods listed here and see if some are unnecessary based on UserRepository.